bundles/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/internal/discovery/FoobotAccountDiscoveryService.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.foobot.internal.discovery;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.openhab.binding.foobot.internal.FoobotApiException;
+import org.openhab.binding.foobot.internal.FoobotBindingConstants;
+import org.openhab.binding.foobot.internal.FoobotHandlerFactory;
+import org.openhab.binding.foobot.internal.handler.FoobotAccountHandler;
+import org.openhab.binding.foobot.internal.handler.FoobotDeviceHandler;
+import org.openhab.binding.foobot.internal.json.FoobotDevice;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FoobotAccountDiscoveryService} is responsible for starting the discovery procedure
+ * that retrieves Foobot account and imports all registered Foobot devices.
+ *
+ * @author George Katsis - Initial contribution
+ * @author Hilbrand Bouwkamp - Completed implementation
+ */
+@NonNullByDefault
+public class FoobotAccountDiscoveryService extends AbstractDiscoveryService
+        implements DiscoveryService, ThingHandlerService {
+
+    private static final int TIMEOUT_SECONDS = 5;
+
+    private final Logger logger = LoggerFactory.getLogger(FoobotAccountDiscoveryService.class);
+
+    private @Nullable FoobotAccountHandler handler;
+    private @NonNullByDefault({}) ThingUID bridgeUID;
+
+    public FoobotAccountDiscoveryService() {
+        super(FoobotHandlerFactory.DISCOVERABLE_THING_TYPE_UIDS, TIMEOUT_SECONDS);
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        retrieveFoobots();
```

Comment:
Hm, this is doing a one-time discovery, not a (continuous) background discovery. I would claim that this DiscoveryService should not at all claim that it offers background discovery (add a `false` to the super call in the constructor).